src/pyinfrabox/infrabox/__init__.py

```diff
@@ -537,7 +538,7 @@ def validate_json(d):
             if job_name == parent_name:
                 raise ValidationError(path, "Job '%s' may not depend on itself" % parent_name)
 
-            if parent_name not in jobs:
+            if parent_name not in all_job_names:
                 raise ValidationError(path + ".depends_on", "Job '%s' not found" % parent_name)
```

Comment:
now we need an additional check that a job does not depend on itself.

Reply:
This situation is checked in line 537

Reply:
I think we should check  if there is a loop in dependency graph

Reply:
yes it's already checked, but you are  right cyclic dependencies have to be checked